src/core/hle/kernel/hle_ipc.cpp

```diff
@@ -16,6 +16,47 @@
 
 namespace Kernel {
 
+class HLERequestContext::ThreadCallback : public Kernel::WakeupCallback {
+
+public:
+    ThreadCallback(std::shared_ptr<HLERequestContext> context_,
+                   std::shared_ptr<HLERequestContext::WakeupCallback> callback_)
+        : context(context_), callback(callback_) {}
```

Comment:
```suggestion
        : context(std::move(context_)), callback(std::move(callback_)) {}
```

Otherwise you have two unnecessary atomic reference count increments and decrements here.